upload/upload_registrar.py

```diff
@@ -76,61 +40,93 @@ def parse_file(filename, verbose = False):
-                middle_name = parsed_line[3],
-                uid = int(parsed_line[4]),
-                gender = 0 if parsed_line[5] == "M" else 1,
-                birthday = datetime.datetime
-                    .strptime(parsed_line[6], "%d-%b-%Y").date(),
-                msc = int(re.findall(r"\d+", parsed_line[10])[0]),
-                phone = parsed_line[11],
-                email = parsed_line[12],
-                address = "".join(["" if line == None else line for line in
+            new_member = {
+                "last_name": parsed_line[1],
+                "first_name": parsed_line[2],
+                "middle_name": parsed_line[3],
+                "uid": parsed_line[4],
+                "gender": 0 if parsed_line[5] == "M" else 1,
+                "birthday": datetime.datetime
+                    .strptime(parsed_line[6].lower(), "%d-%b-%Y").date(),
+                "msc": int(re.findall(r"\d+", parsed_line[10])[0]),
+                "phone": parsed_line[11],
+                "email": parsed_line[12],
+                "address": "".join(["" if line == None else line for line in
                     parsed_line[13:15]]),
-                city = parsed_line[16],
-                state = parsed_line[17],
-                zip = parsed_line[18],
-                country = "United States" if parsed_line[19] == None
+                "city": parsed_line[16],
+                "state": parsed_line[17],
+                "zip": parsed_line[18],
+                "country": "United States" if parsed_line[19] == None
                     else parsed_line[19]
-            )
+            }
             parsed_data.append(new_member)
         if verbose:
-            print "Loaded and parsed %d students." % len(parsed_data)
+            print "Loaded and parsed {} students.".format(len(parsed_data))
         return parsed_data
     
 # connect to the desired database
-def upload_to_db(student_data, database, verbose = False):
+def upload_to_db(member_data, database, verbose = False):
     """
     Establish a connection to the desired database and upload student data to
     it. Data is an array of member objects, where each member object is a
     member parsed from the registrar file.
     """
     if verbose:
-        print "Connecting to \"%s\" database." % database
+        print "Connecting to \"{}\" database.".format(database)
     db_conn = db.DB(database)
-    Session = sessionmaker(bind = db_conn.engine)
-    session = Session()
-    if verbose:
-        print "Connected."
     
     if verbose:
-        print "Uploading students."
-    for student in student_data:
-        session.add(student)
-    try:
-        session.commit()
-    except IntegrityError as e:
-        print e
+        print "Connected."
+        print "Staging changes..."
+        
+    transaction = db_conn.db.begin()
+        
+    for member in member_data:
+        cmd = ("INSERT INTO members (last_name, first_name, middle_name, uid, "
```

Comment:
You should parameterize this query. See https://github.com/ASCIT/donut/blob/master/donut/modules/auth/helpers.py#L24 for an example